src/Event.cs

```diff
@@ -254,7 +254,7 @@ public void UnsubscribeThen(Action<SubscriptionEventContext>? onEnded)
             {
                 throw new Exception("Cannot unsubscribe from inactive subscription.");
             }
-            if (onEnded != null)
+            if (this.onEnded != null)
```

Comment:
it also looks like we have a separate "bug" where calling `Unsubscribe` twice won't actually trigger the exception below? (Because `this.onEnded` will be `null` if we called `Unsubscribe` since that triggers this function with a `null` param) Is that.. an issue..?

Reply:
GOOOD point